Frontend/ui-customer/src/Screens/ZooTicketBookingFlow/TicketBookingScreen/Controller.purs

```diff
@@ -133,68 +163,251 @@ eval (PaymentStatusAction status) state =
 
 eval (RefreshStatusAC (PrimaryButton.OnClick)) state = exit $ RefreshPaymentStatus state
 
+eval (SelectDestination selCategory) state = do
+  let modifiedServicesData = map (modifyDestinationInfo) state.data.servicesInfo
+      finalAmount = calculateTotalAmount state modifiedServicesData
+  continue state {data {totalAmount = finalAmount, servicesInfo = modifiedServicesData}}
+  where
+    modifyDestinationInfo service = if service.id == selCategory.ticketID then service {businessHours = map (modifyBH service.selectedBHid) service.businessHours} else service
+    modifyBH selectedBHid bh = if (Just bh.bhourId == selectedBHid) then bh { categories = map modifyCategory bh.categories} else bh
+    modifyCategory category = category { isSelected = selCategory.categoryId == category.categoryId}
+
+eval (SelectSlot ticketId slot) state = do
+  let modifiedServicesData = map (modifySlotInfo) state.data.servicesInfo 
+      finalAmount = calculateTotalAmount state modifiedServicesData
+  continue state {data {totalAmount = finalAmount, servicesInfo = modifiedServicesData}}
+  where
+    modifySlotInfo service =  if service.id == ticketId then service { selectedSlot = Just slot.slot, selectedBHid = Just slot.bhourId, businessHours = map (modifyBH service.selectedBHid) service.businessHours} else service
+    modifyBH selectedBHid bh =  bh {categories = map (initCategories bh) bh.categories} 
+    initCategories bh cat = cat {isSelected = if length bh.categories > 1 then false else true, peopleCategories = map initPeopleCategories cat.peopleCategories}
+    initPeopleCategories pc = pc {currentValue = 0}
+
+eval (OpenGoogleMap lat long) state = do
+  let dstLat = fromMaybe 0.0 lat
+      dstLong = fromMaybe 0.0 long
+  updateAndExit state $ GoToOpenGoogleMaps state dstLat dstLong
+
+eval (Copy text) state = continueWithCmd state [ do 
+    _ <- pure $ JB.copyToClipboard text
+    _ <- pure $ JB.toast (getString COPIED)
```

Comment:
`void $ pure $ JB.toast $ getString COPIED`
This will look cleaner

Reply:
done